flink-python/src/test/java/org/apache/flink/table/runtime/operators/python/aggregate/arrow/batch/BatchArrowPythonGroupWindowAggregateFunctionOperatorTest.java

```diff
@@ -0,0 +1,327 @@
+		expectedOutput.add(new StreamRecord<>(
+			newRow(true, "c1", 2L, TimestampData.fromEpochMillis(10000L), TimestampData.fromEpochMillis(20000L))));
+
+		assertOutputEquals("Output was not correct.", expectedOutput, testHarness.getOutput());
+
+		testHarness.close();
+
+		expectedOutput.add(new StreamRecord<>(
+			newRow(true, "c2", 3L, TimestampData.fromEpochMillis(-5000L), TimestampData.fromEpochMillis(5000L))));
+		expectedOutput.add(new StreamRecord<>(
+			newRow(true, "c2", 3L, TimestampData.fromEpochMillis(0L), TimestampData.fromEpochMillis(10000L))));
+
+		assertOutputEquals("Output was not correct.", expectedOutput, testHarness.getOutput());
+	}
+
+	@Test
+	public void testFinishBundleTriggeredByTime() throws Exception {
+		Configuration conf = new Configuration();
+		conf.setInteger(PythonOptions.MAX_BUNDLE_SIZE, 10);
+		conf.setLong(PythonOptions.MAX_BUNDLE_TIME_MILLS, 1000L);
+		OneInputStreamOperatorTestHarness<RowData, RowData> testHarness = getTestHarness(conf);
+
+		long initialTime = 0L;
+		ConcurrentLinkedQueue<Object> expectedOutput = new ConcurrentLinkedQueue<>();
+
+		testHarness.open();
+
+		testHarness.processElement(new StreamRecord<>(newBinaryRow(true, "c1", "c2", 0L, 0L), initialTime + 1));
+		testHarness.processElement(new StreamRecord<>(newBinaryRow(true, "c1", "c4", 1L, 6000L), initialTime + 2));
+		testHarness.processElement(new StreamRecord<>(newBinaryRow(true, "c1", "c6", 2L, 10000L), initialTime + 3));
+		testHarness.processElement(new StreamRecord<>(newBinaryRow(true, "c2", "c8", 3L, 0L), initialTime + 3));
+		assertOutputEquals("FinishBundle should not be triggered.", expectedOutput, testHarness.getOutput());
+
+		testHarness.setProcessingTime(1000L);
+		expectedOutput.add(new StreamRecord<>(
+			newRow(true, "c1", 0L, TimestampData.fromEpochMillis(-5000L), TimestampData.fromEpochMillis(5000L))));
+		expectedOutput.add(new StreamRecord<>(
+			newRow(true, "c1", 0L, TimestampData.fromEpochMillis(0L), TimestampData.fromEpochMillis(10000L))));
+		expectedOutput.add(new StreamRecord<>(
+			newRow(true, "c1", 1L, TimestampData.fromEpochMillis(5000L), TimestampData.fromEpochMillis(15000L))));
+		expectedOutput.add(new StreamRecord<>(
+			newRow(true, "c1", 2L, TimestampData.fromEpochMillis(10000L), TimestampData.fromEpochMillis(20000L))));
+		assertOutputEquals("Output was not correct.", expectedOutput, testHarness.getOutput());
+
+		testHarness.close();
+
+		expectedOutput.add(new StreamRecord<>(
+			newRow(true, "c2", 3L, TimestampData.fromEpochMillis(-5000L), TimestampData.fromEpochMillis(5000L))));
+		expectedOutput.add(new StreamRecord<>(
+			newRow(true, "c2", 3L, TimestampData.fromEpochMillis(0L), TimestampData.fromEpochMillis(10000L))));
+
+		assertOutputEquals("Output was not correct.", expectedOutput, testHarness.getOutput());
+	}
+
+	@Test
+	public void testWatermarkProcessedOnFinishBundle() throws Exception {
```

Comment:
BatchArrowPythonGroupWindowAggregateFunctionOperator is running in batch and so I guess we don't need to consider watermark?